### ISSUE TITLE
Improve handling of DB storage full behavior on metered networks

### ISSUE THREAD

**Issue by kindbe** [microsoft/cpp_client_telemetry#431]
Some comments from @maxgolov in PR #427...

I think we need to follow-up on how it works when the DB is full of Normal pri events on Metered connection. Some arch rework possibly needed in here:

cpp_client_telemetry/lib/system/TelemetrySystem.cpp

Line 192 in 7ca71f8
`tpm.scheduleUpload(0, EventLatency_Normal, true); `
which is called from UploadNow here:

cpp_client_telemetry/lib/api/LogManagerImpl.cpp

Line 346 in 0ecd7cc
` GetSystem()->upload(); `
Basically TelemetrySystem should respect the current network cost, and avoid unnecessarily scheduling normal when on metered. That'd save DB query time.
